nschecker/checker.go

```diff
@@ -40,8 +40,27 @@ func (s State) String() string {
 	return "Unknown state"
 }
 
-func Check(s Source) (State, error) {
-	resp, err := http.Get(s.URL)
+type CheckOption func(*checkOptions)
+
+type checkOptions struct {
+	hc *http.Client
+}
+
+func HTTPClient(hc *http.Client) CheckOption {
+	return func(o *checkOptions) {
+		o.hc = hc
+	}
+}
+
+func Check(s Source, opts ...CheckOption) (State, error) {
```

Comment:
hmm... i'm not sure it's good interface...

I think just passing HTTP.Client is fine. Use DefaultClient if it's nil.